lib/DocumentRegistry.php

```diff
@@ -73,44 +69,22 @@ public function __construct($defaultLocale)
      * @param mixed $document
      * @param NodeInterface $node
      * @param NodeInterface $node
-     * @param null|string $locale
+     * @param string $locale
      *
      * @throws DocumentManagerException
      */
-    public function registerDocument($document, NodeInterface $node, $locale = null)
+    public function registerDocument($document, NodeInterface $node, $locale)
     {
-        if (null === $locale) {
-            $locale = $this->defaultLocale;
-        }
-
         $oid = $this->getObjectIdentifier($document);
         $uuid = $node->getIdentifier();
 
         // do not allow nodes without UUIDs or reregistration of documents
-        $this->validateDocumentRegistration($document, $node, $oid, $uuid);
+        $this->validateDocumentRegistration($document, $locale, $node, $oid, $uuid);
 
         $this->documentMap[$oid] = $document;
         $this->documentNodeMap[$oid] = $uuid;
         $this->nodeMap[$node->getIdentifier()] = $node;
-        $this->nodeDocumentMap[$node->getIdentifier()] = $document;
-        $this->documentLocaleMap[$oid] = $locale;
-    }
-
-    /**
-     * Update the locale of the given document and store the originally
-     * requested locale.
-     *
-     * The originally requested locale should be reset when a HYDRATE event
-     * is caused by the user (and not internally when loading dependencies).
-     *
-     * @param object $document
-     * @param string $locale
-     * @param null|string $originalLocale
-     */
-    public function updateLocale($document, $locale, $originalLocale = null)
-    {
-        $oid = $this->getObjectIdentifier($document);
-        $this->originalLocaleMap[$oid] = $originalLocale;
+        $this->nodeDocumentMap[sprintf('%s-%s', $node->getIdentifier(), $locale)] = $document;
```

Comment:
Any chances of naming collisions here? Maybe use a different sign than `-` to avoid confusion with the dashes from the UUID? `_` comes to mind.